src/Aspire.Hosting/ApplicationModel/ContainerExecutableResource.cs

```diff
@@ -9,8 +9,8 @@ namespace Aspire.Hosting.ApplicationModel;
 /// <summary>
 /// Executable resource that runs in a container.
 /// </summary>
-internal class ContainerExecutableResource(string name, ContainerResource containerResource, string command, string? workingDirectory)
-    : Resource(name), IResourceWithEnvironment, IResourceWithArgs, IResourceWithEndpoints, IResourceWithWaitSupport
+public class ContainerExecutableResource(string name, ContainerResource containerResource, string command, string? workingDirectory)
+    : Resource(name), IResourceWithEnvironment, IResourceWithArgs, IResourceWithEndpoints, IResourceWithWaitSupport, IResourceWithParent<ContainerResource>
```

Comment:
Regarding `IResourceWithEndpoints` I think we should remove it for now. I think realistically it is the parent resource that defines the endpoints for `ContainerExecutableResource`. I'm not 100% sure on this but its the most conservative option for now.

In theory we could create a ContainerExecutableResource that starts some process that listens on a port - but the problem is I don't think we can forward another port to the container when we run `docker exec` under the covers - so the port would have had already had to been forwarded on the parent.

@karolz-ms thoughts on this?

Reply:
totally agree, forgot to remove it.

Reply:
Yeah, you can’t publish a new port after a container is already running; you could make it accessible over the network to other containers, but there’s no way to model that with an endpoint currently.